app/services/nla/citations/wikimedia_citation_service.rb

```diff
@@ -106,10 +116,24 @@ def build_language
         end
       end
 
-      def build_pi
-        pi = @document.pi
-        if pi.present?
-          " | url=#{pi.first}\n"
+      def build_access_date
+        date = Time.zone.today.strftime("%d %B %Y")
+        if date.present?
+          " | access-date=#{date}\n"
+        end
+      end
+
+      def build_persistent_url
+        persistent_url = @document.id
+        if persistent_url.present?
+          " | url=https://nla.gov.au/nla.cat-vn#{persistent_url}\n"
+        end
+      end
+
+      def build_via
+        via = "National Library of Australia"
+        if via.present?
+          " | via=#{via}\n"
         end
```

Comment:
Don't think these functions are needed as the way they are coded the present() method will always be true, two with fixed strings and on with Solr document id which must be set to have existed in Solr. I'd just either return the string or just merge the concatenation with the code above it.